src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,45 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder getSalaryOutput = new StringBuilder();
+        int[] salary = new int[names.length];
+        final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        final LocalDate dateFromFormated = LocalDate.parse(dateFrom, formatter);
+        final LocalDate dateToFormated = LocalDate.parse(dateTo, formatter);
+        getSalaryOutput.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+        for (String dataArray : data) {
```

Comment:
Using 'dataArray' as a variable name is not informative. Consider renaming it to something like 'record' or 'dataEntry' to better reflect the data it represents.